### DISCUSSION TITLE
Shopify Script discounts not showing up in Cart discountAllocations

### DISCUSSION THREAD

**Discussion by mjvotaw** [Shopify/storefront-api-feedback#28] [Help] [answered]
Hello!

So I've got a Line Item script published on my store, and I'm trying to implement the new Cart Storefront features. The script gets correctly applied on the web store, but isn't getting applied if I create a cart through the Storefront api. If I get the checkoutUrl and start a web checkout, the discounts show up correctly there. Is this still just a work in progress?

**Comment by morganmccunn** [comment, marked as answer]
Hey @mjvotaw ! Yep this is a current known limitation of the cart. [![alt](https://screenshot.click/08-25-p6udw.jpg)](https://screenshot.click/08-25-p6udw.jpg)

We are actively working with a few teams to get this working in the cart, however it's quite nuanced and we expect this to take at least another quarter. The scripts will continue to get applied at the checkout for the final pricing, so one alternative would be to make some changes on the client-side to reflect this in the cart in the interim!

**Reply by mjvotaw** [reply]
Oof, right there in the documentation too. Thanks for the update about expected timing, that's really helpful to know.
